flask_session_captcha/__init__.py

```diff
@@ -1,97 +1,159 @@
-        elif session_type == "sqlalchemy":
-            # I have to do this as of version 0.3.1 of flask-session if using sqlalchemy as the session type in order to create the initial database.
-            # Flask-sessionstore seems to have the same problem.
-            app.session_interface.db.create_all()
-
-    def generate(self):
+    def generate(self,*args, **kwargs) -> str:
         """
-        Generates and returns a numeric captcha image in base64 format. 
-        Saves the correct answer in `session[self.session_key]`
-        Use later as:
+            generate captcha with given flags
+
+            :param:
+                numeric: if True generate captcha with numeric only
+                alphabet: if True generate captcha with alphabet
+                punctuation if True generate captcha with punctuation symbols
+
+            if both alphabet and numeric set to True this function generate captcha that contain both alphabet and numeric
 
-        src = captcha.generate()
-        <img src="{{src}}">
         """
-        answer = self.rand.randrange(self.max)
-        answer = str(answer).zfill(self.digits)
+
+        if not kwargs.get("numeric"):
+            numeric = self.include_numeric
+        if not kwargs.get("alphabet"):
+            alphabet = self.include_alphabet
+        if not kwargs.get("punctuation"):
+            punctuation = self.include_punctuation
+
+        # single mode captcha options
+        answer = []
+        if numeric and not alphabet and not punctuation:
+            answer += self.random_number(length=self.length)
+        if alphabet and not numeric and not punctuation:
+            answer += self.random_alphabet(length=self.length)
+        if punctuation and not numeric and not alphabet:
+            answer += self.random_punctuation(length=self.length)
+
+        if len(answer) == 0:  # mix captcha mode
+            selected = {}
+            if punctuation:
+                selected["punctuation"] = self.random_punctuation
+            if numeric:
+                selected["numeric"] = self.random_number
+            if alphabet:
+                selected["alphabet"] = self.random_alphabet
+
+            total = sum([alphabet, numeric, punctuation])
+            each_round = self.length // total
+            for each in selected:
+                answer += (selected[each](length=each_round))
+
+            answer += (self.random_alphabet(length=(self.length - len(answer))))
+            self.shuffle_list(answer)
+
+        answer = "".join(answer)
         image_data = self.image_generator.generate(answer)
         base64_captcha = base64.b64encode(
             image_data.getvalue()).decode("ascii")
-        logging.debug('Generated captcha with answer: ' + answer)
+        self.Logger.info(f'Captcha Generated:\nKey:{answer}\tAnswer:{answer}')
```

Comment:
Typically `info`  level logs should not reveal sensitive information.

Reply:
yes i forget to change it